### ISSUE TITLE
Shapes, legend symbols, and `showtext`

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/TLF-Library#449]
Since one of the shapes updates my plots are messed up and I cannot figure out how to get it right. Three examples:

1. Scatter with  `pentagon` as symbol

`addScatter(x = 1:10, y = rlnorm(10), shape = Shapes$pentagon, color = "blue", size = 10)`

2. Scatter with `sheep` as symbol

`addScatter(x = 1:10, y = rlnorm(10), shape = Shapes$sheep, color = "blue", size = 10)`

3. Simple time profile with `ospsuite`

```
library("ospsuite")
ds1 <- DataSet$new("ds1")
ds1$setValues(c(1,2), c(1,2))

ds2 <- DataSet$new("ds2")
ds2$setValues(c(1,2), c(3,4))

dc <- DataCombined$new()

dc$addDataSets(c(ds1, ds2))

plotIndividualTimeProfile(dc)
```

## TLF 1.5.125
Before adding new shapes, this worked like this:

1. ![image](https://user-images.githubusercontent.com/25903040/229081218-be07a557-333c-469b-a18a-059d2f1a2f4c.png)
2. ![image](https://user-images.githubusercontent.com/25903040/229081292-fa7a02fd-1fef-4d45-97b0-e08abe7e2529.png)
3. ![image](https://user-images.githubusercontent.com/25903040/229081426-2b184580-7480-4eab-845c-9d1261f9e811.png)

So there were no pentagons or sheep, but the time profile looked as expected.

However, manually providing the UTF code for pentagon or sheep works, so actually everything worked as expected:

`tlf::addScatter(x = 1:10, y = rlnorm(10), shape = "\u2b1f", color = "blue", size = 10)`
![image](https://user-images.githubusercontent.com/25903040/229082093-b8991c4d-8433-4be1-a5b0-5fe9a42c11fe.png)

`tlf::addScatter(x = 1:10, y = rlnorm(10), shape = "\ud83d\udc11", color = "blue", size = 10)`
![image](https://user-images.githubusercontent.com/25903040/229084532-e450e62a-9dd9-4998-81c9-cb15fbe0c745.png)

## TLF 1.5.135 (latest develop)

With the latest develop, without `showtext` installed:

Pentagons and sheep as expected (see previous plots), but the time profile is now:

![image](https://user-images.githubusercontent.com/25903040/229085802-74a4271f-2185-4e90-b95d-2173bfbde11a.png)
Mind the symbol size and broken legend entries!

### Installing `showtext`
1. 
![image](https://user-images.githubusercontent.com/25903040/229086219-f15ed5fc-57fe-4cf0-8955-604aa82a45be.png)
The font is different (I actually do not like this font, text looked better in the previous versions)

2. 
![image](https://user-images.githubusercontent.com/25903040/229086447-03a35963-cb9c-47ba-b71a-c0734a258a65.png)
What happened to the sheep? 👎 I want the old sheep back!

3. 
![image](https://user-images.githubusercontent.com/25903040/229086825-de38a4a9-08b9-464f-ad36-fe8b91083aee.png)
Text is ugly, everything seems to be "bold" (including lines etc), legend still broken.

##  TLF 1.5.135 (latest develop) without adding the font
Commenting out

```
  sysfonts::font_add(
      family = "Symbola",
      regular = system.file("extdata", "Symbola.ttf", package = "tlf")
    )
```

1. 
![image](https://user-images.githubusercontent.com/25903040/229088106-fdbdaf3e-2092-4a81-a0df-47ab58457dbf.png)
Now it looks like what @Yuri05 had here https://github.com/Open-Systems-Pharmacology/TLF-Library/issues/433

2. Same for the sheep, no sheep.
![image](https://user-images.githubusercontent.com/25903040/229088521-3c4fdcfa-8145-4a99-8acf-8435ae380608.png)

Still broken.

Bottom line - can we somehow get to the original behavior where everything worked?

**Comment by Yuri05** [comment]
@PavelBal can you try with the last dev (1,5.137)?

**Comment by pchelle** [comment]
@PavelBal ,
The sheep symbol you liked was only available thanks to your computer settings but it was usually rendered differently in other computers (often filled or empty squares) because the font used for the symbols was 1) selected by `ggplot2` and 2) could not be tuned. Which makes it impossible to get consistent symbols across computers and systems.
Another issue is that the symbols leveraged unicode characters that are usually not converted when exporting the plot in svg, postscript or pdf.

To fix these issues, `showtext` seems a rather good way as it allows the installation and use of font files (that can be used to provide consistent shapes) and converts them as images before exporting preventing issues of translation of unicode characters.
PR #442 fixed some issues especially in displayed legend.

The consistency in behaviour with/without `{showtext}` needs further improvement. Especially, most of the new shapes should be removed when `{showtext}` is not available.

**Comment by pchelle** [comment]
Currently, the font used for the shapes is included in the file `Symbola.ttf`.
I found a font that had already most of the shapes we were looking for.

It is however possible to define cool shapes by creating your own font using websites such as https://glyphter.com/.
You can upload there svg images for each letter, number, etc. of the alphabet and then download the corresponding .ttf file.
Then, the shapes could be available to `{tlf}` by updating the dictionary of `Shapes`.
A potential benefit is that you can include shapes that can represent animal models but also organs, administrations, small/large molecules, etc. which could be interesting for sensitivity plots or such.

Below is a test of this potential feature/fix:

<details><summary>Creating font from https://glyphter.com/ </summary>

![image](https://user-images.githubusercontent.com/45343665/229474872-897d0cd8-3e7a-4c18-b23c-15ca86a40a72.png)

</details>

<details><summary>Check how font is rendered </summary>

```r
# Add new font to available font families
sysfonts::font_add(family = "test", regular = "test.ttf")
showtext::showtext_auto()

library(ggplot2)

ggplot(
  data.frame(label = LETTERS[1:8]),
  aes(x=1,y=label,label=label)
) + geom_text(family = "test", size = 10, color = "firebrick")

```

![image](https://user-images.githubusercontent.com/45343665/229477291-7f244775-9d92-4254-a873-2279bb0df250.png)


</details>

**Comment by PavelBal** [comment]
Hehe that's cool.

**Comment by PavelBal** [comment]
With the latest dev, the result is as expected, I guess (sheep are not as beautiful as they used to be, but I guess I will have to live with it).
